textile/chat-features.textile

```diff
@@ -411,36 +411,56 @@ Presence allows chat room users to indicate to others that they're online, as we
-* @(CHA-T3)@ @[Testable]@ Users may configure a timeout interval for when they are typing. This configuration is provided as part of the @RoomOptions@ @typing.timeoutMs@ property, or idiomatic equivalent. The default is 5000ms.
-* @(CHA-T4)@ Users may indicate that they have started typing.
-** @(CHA-T4a)@ @[Testable]@ If typing is not already in progress, per explicit cancellation or the timeout interval in (@CHA-T3@), then a new typing session is started.
-*** @(CHA-T4a1)@ @[Testable]@ When a typing session is started, the client is entered into presence on the typing channel.
-*** @(CHA-T4a2)@ @[Testable]@ When a typing session is started, a timeout is set according to the @CHA-T3@ timeout interval. When this timeout expires, the typing session is automatically ended by leaving presence.
-** @(CHA-T4b)@ @[Testable]@ If typing is already in progress, the @CHA-T3@ timeout is extended to be @timeoutMs@ from now.
-* @(CHA-T5)@ Users may indicate that they have stopped typing.
-** @(CHA-T5a)@ @[Testable]@ If typing is not in progress, this operation is no-op.
-** @(CHA-T5b)@ @[Testable]@ If typing is in progress, he @CHA-T3@ timeout is cancelled. The client then leaves presence.
+For the purpose of this section, an @ephemeral message@ is understood to mean a regular message where @message.extras.ephemeral@ is set to @true@.
+
+* @(CHA-T1)@ This specification point has been removed. It was superseded by @CHA-T8@.
+* @(CHA-T2)@ This specification point has been removed. It was replaced by @CHA-T9@.
+* @(CHA-T8)@ Typing Indicators for a Room are exposed on the main room channel, in the format @<roomId>::$chat@. For example, if your room id is @my-room@ then the channel will be @my-room::$chat@.
+* @(CHA-T9)@ @[Testable]@ Users may retrieve a list of the currently typing client IDs.
+* @(CHA-T10)@ @[Testable]@ Users may configure a heartbeat interval (the no-op period for @typing.keystroke@ when the heartbeat timer is set active at @CHA-T4a4@). This configuration is provided at the @RoomOptions.typing.heartbeatThrottleMs@ property, or idiomatic equivalent. The default is 10000ms.
+** @(CHA-T10a)@ @[Testable]@ A grace period shall be set by the client (the grace period on the @CHA-T10@ heartbeat interval when receiving heartbeats). The value shall be set to 2000ms.
+*** @(CHA-T10a1)@ @[Testable]@ If a @typing.start@ event is not received within this period, the client shall assume that the user has stopped typing. For example, if the client has not received a @typing.start@ event within 12000ms of the last heartbeat (10s heartbeat interval plus 2s grace period), the client shall assume that the user has stopped typing.
+* @(CHA-T3)@ This specification point has been removed. It was replaced by @CHA-T10@.
+* @(CHA-T4)@ Users may indicate that they have started typing using the @keystroke@ method.
+** @(CHA-T4d)@ @[Testable]@ If the channel state is not @ATTACHED@ or @ATTACHING@, the operation shall throw an @ErrorInfo@ with code @40000@.
```

Comment:
Seems, this spec will be updated in the future as per internal slack discussion -> https://ably-real-time.slack.com/archives/C02NY1VT3LY/p1742981576970129

Reply:
For, now we can ignore/ keep as is